### ISSUE TITLE
There's a problem:  Jacobian shape: torch.Size([64, 10, 0]).

### ISSUE THREAD

**Issue by fly2tortoise** [ondratybl/VKDNW#2]
Dear Professor, I am calculating **VKDNW on the 201** space. I found that the returned dimension is incorrect when calculating the **Jacobian value**. How should this part be fixed? **Thank you.**

def get_fisher(model, input, use_logits=True, params_grad_len=256, p=0.):

    model.eval()

    acobian = get_jacobian_index(model, input, p, params_grad_len)
    print("Jacobian shape:", jacobian.shape)  # Jacobian shape: torch.Size([64, 10, 0]).

    if not use_logits:
        jacobian = torch.matmul(cholesky_covariance(model(input)[1]), jacobian).detach()

    fisher = torch.mean(torch.matmul(torch.transpose(jacobian, dim0=1, dim1=2), jacobian), dim=0).detach()

    del jacobian
    gc.collect()
    torch.cuda.empty_cache()
    return fisher

**Comment by ondratybl** [comment]
Hi! This will happen if you pass in a model without any trainable parameters. Are you sure you did not pass there the model e.g. with all connections being zeros?

**Comment by fly2tortoise** [comment]
Dear Professor, I fixed the issue with tensor 0 today. I found that it was mainly caused by the parameter p.

```
p = 1
fisher_prob = get_fisher(model, input_, use_logits=False, params_grad_len=params_grad_len, p=p)
```

However, when I set p to 10 on NB201, the KT of 3000 architectures is 0.447, which is still somewhat different from the 0.8 in the paper. Where should I make changes to reproduce the KT value that surpasses AZNAS in the paper?

Thank you very much for your reply.
